src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineView.xaml.cs

```diff
@@ -0,0 +1,175 @@
+
+        private int StartTrackingView(IVsTextView textView)
+        {
+            var wpfTextView = _editorAdaptersFactoryService.GetWpfTextView(textView);
+            if (wpfTextView is null)
+                return VSConstants.E_FAIL;
+
+            _trackedTextViews.Add(textView, wpfTextView);
+
+            // In the split window case, there's two views (each with its own caret position) but only one text buffer.
+            // Subscribe to caret position changes once per view.
+            wpfTextView.Caret.PositionChanged += Caret_PositionChanged;
+
+            return VSConstants.S_OK;
+        }
+
+        private void ExpandAll(object sender, RoutedEventArgs e)
+            => _viewModel.EnqueueExpandOrCollapse(ExpansionOption.Expand);
+
+        private void CollapseAll(object sender, RoutedEventArgs e)
+            => _viewModel.EnqueueExpandOrCollapse(ExpansionOption.Collapse);
+
+        private void SearchBox_TextChanged(object sender, EventArgs e)
+        {
+            var newText = SearchBox.Text ?? string.Empty;
+            _viewModel.EnqueueFilter(newText);
+        }
+
+        private void SortByName(object sender, EventArgs e)
+            => UpdateSort(SortOption.Name, nameof(DocumentSymbolDataViewModel.Name));
+
+        private void SortByOrder(object sender, EventArgs e)
+            => UpdateSort(SortOption.Location, nameof(DocumentSymbolDataViewModel.StartPosition));
+
+        private void SortByType(object sender, EventArgs e)
+            => UpdateSort(SortOption.Type, nameof(DocumentSymbolDataViewModel.SymbolKind), nameof(DocumentSymbolDataViewModel.Name));
+
+        private void UpdateSort(SortOption sortOption, params string[] sortProperties)
+        {
+            // Log which sort option was used
+            Logger.Log(sortOption switch
+            {
+                SortOption.Name => FunctionId.DocumentOutline_SortByName,
+                SortOption.Location => FunctionId.DocumentOutline_SortByOrder,
+                SortOption.Type => FunctionId.DocumentOutline_SortByType,
+                _ => throw new NotImplementedException(),
+            }, logLevel: LogLevel.Information);
+
+            var view = ((CollectionViewSource)FindResource("DocumentSymbolItems")).View;
+            view.SortDescriptions.Clear();
+            foreach (var sortProperty in sortProperties)
+            {
+                view.SortDescriptions.Add(new SortDescription(sortProperty, ListSortDirection.Ascending));
+            }
+
+            _viewModel.SortOption = sortOption;
+            view.Refresh();
+        }
+
+        /// <summary>
+        /// When a symbol node in the window is clicked, move the caret to its position in the latest active text view.
+        /// </summary>
+        private void SymbolTree_MouseDown(object sender, EventArgs e)
+        {
+            if (sender is StackPanel panel && panel.DataContext is DocumentSymbolDataViewModel symbol)
+            {
+                _isNavigating = true;
```

Comment:
1. definitely a little oogy about this, especially around the idea of someone just quickly clicking on N items.
2. feels odd that thsi is MouseDown.  how does thsi work for someone navigating this tree with keyboard?  Shouldn't this just be triggered based on an item being selected, rather than mousedown?